doc/programming/language/statements.rst

```diff
@@ -269,9 +269,9 @@ exception.
    [else
         BLOCK]
 
-A while-loop that executed ``BLOCK`` for as long as the boolean
-``COND`` evaluates to true. The second form initializes a new local
-variable ``IDENT`` with ``EXPR``, and makes it available inside both
-``COND`` and the body. If an ``else`` block is specified, that will
-executes once the ``COND`` evaluates to false at the loop beginning.
+``while`` introduces a loop that executed ``BLOCK`` for as long as the
+boolean ``COND`` evaluates to true. The second form initializes a new
+local variable ``IDENT`` with ``EXPR``, and makes it available inside
+both ``COND`` and ``BLOCK``. If an ``else`` block is specified, it will
+execute one time once ``COND`` evaluates to false at the loop beginning.
```

Comment:
it will execute one time once
should be
it will execute one time if

Reply:
I am not a native speaker, but I parse 

>  it will execute one time once ``COND`` evaluates to false at the loop beginning.

as _at the point it time where `! COND` execute it_, but

>  it will execute one time if ``COND`` evaluates to false at the loop beginning.

at least has a ring of _if at the beginning `! COND` execute it_, i.e., either execute the loop at least once or execute the `else` block.

At least from looking at the tests the implementation does the first of these.

Reply:
The piece that we are revising is the trailing half of this sentence: If an ``else`` block is specified, it will execute one time once ``COND`` evaluates to false at the loop beginning.
So the more accurate statement is:
If an ``else`` block is specified, it will execute one time if ``COND`` evaluates to false at the loop beginning.

I.e. It will never execute at all,  if ``COND`` evaluates to true at the loop beginning.

"...once ``COND`` evaluates to ..." is linguistically a usage for a situation which is sure to eventually happen.

Reply:
> "...once COND evaluates to ..." is linguistically a usage for a situation which is sure to eventually happen.

And that is exactly what is happening here. The `else` block will always be executed unless the loop terminates via `break`. As an example, [this loop](https://github.com/zeek/spicy/blob/3bfb6cf3a8790cb154eac35e445aa733c77a619a/tests/spicy/statements/while.spicy#L30-L36) is executed like [this](https://github.com/zeek/spicy/blob/3bfb6cf3a8790cb154eac35e445aa733c77a619a/tests/Baseline/spicy.statements.while/output#L15-L21). Note that the next loop in that test is terminated via `break` so the `else` block is not run.

Reply:
If I have misunderstood whether the else is exclusively ever executed *instead of* ever doing the body, then my suggestion for clarification is clarity but in the wrong direction.

if the else is nearly always executed, one time, after zero-or-more loops through the body, then we should wholesale rewrite the sentence to capture that. Since if the else is not executed at all, when a break occurs.

Reply:
Yeah, looking at https://github.com/zeek/spicy/blob/3bfb6cf3a8790cb154eac35e445aa733c77a619a/tests/Baseline/spicy.statements.while/output#L15-L21 I had formed an inaccurate perception from the pre-existing text: ... it will
execute one time once ``COND`` evaluates to false at the loop beginning.

To my ear, "at the loop beginning" is a one-time moment, just prior to the first evaluation of COND.

Reply:
Okidoki, I think that resolves this.

Reply:
If we are implementing a construct which is: "zero-or-more loops through the body, then perform a statement" we should term that statement as `then` rather than as `else`. Python's statement `finally` is another good choice, though Python's `finally` does execute even when break is used to prematurely drop out from the loop.

Reply:
I doubt that there is good justification for having a construct with semantics:
This statement will not be executed after "zero-or-more loops through the body, if one of those is a conditional to drop out from the loop based upon a conditional other than the COND". 

The use of two distinct conditionals is usually for clarity. It will not be common that an author needs to execute the "finally" kind of statement if the COND happened, and doesn't need to execute that, if execution fell out of the loop at the within-loop conditional. You could just always have that statement, with no else, and it always executes subsequent to the loop.

```
while (idx--) {
   if (isRepresentable(charInSomeEncoding) {
       print charInSomeEncoding.representation();
   else
       break;
}
print "\n";
```

Reply:
I agree and filed #644.